<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Step2 Step-by-step network construction and module detection</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Step2 Step-by-step network construction and module detection</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>wgcna/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210814code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210814)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210814code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210814)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong4ffc582"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 4ffc582 </a>
</p>
</div>
<div id="strongRepositoryversionstrong4ffc582" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 4ffc582. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  05.-Network-visualization-using-WGCNA-functions.Rmd
    Untracked:  analysis/02.-step-by-step-network-construction-and-module-detection.Rmd

Unstaged changes:
    Deleted:    CytoscapeInput-edges-brown-red.txt
    Deleted:    CytoscapeInput-nodes-brown-red.txt
    Modified:   Untitled.R
    Deleted:    VisANTInput-brown-top30.txt
    Deleted:    VisANTInput-brown.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/02. step-by-step network construction and module detection.Rmd</code>) and HTML (<code>docs/02. step-by-step network construction and module detection.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
44f102b
</td>
<td>
lily123920
</td>
<td>
2021-08-15
</td>
<td>
change project name
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="环境配置" class="section level3">
<h3>环境配置</h3>
<pre class="r"><code>library(WGCNA)</code></pre>
</div>
<div id="准备分析数据" class="section level3">
<h3>准备分析数据</h3>
<pre class="r"><code>lnames = load(file = &quot;./data/FemaleLiver-01-dataInput.RData&quot;)
lnames</code></pre>
<pre><code>[1] &quot;datExpr&quot;   &quot;datTraits&quot;</code></pre>
</div>
<div id="一选择软阈值β网络拓扑分析" class="section level3">
<h3>一、选择软阈值β：网络拓扑分析</h3>
<p>更多关于β的介绍参考文献：A general framework for weighted gene co-expression network analysis. Statistical Applications in Genetics and Molecular Biology,2005.</p>
<p>这里主要介绍通过pickSoftThreshold函数确定软阈值过程。</p>
<div id="拓扑网络分析" class="section level5">
<h5>1. 拓扑网络分析</h5>
<pre class="r"><code># 选择一组软阈值
powers = c(c(1:10), seq(from = 12, to=20, by=2))
powers</code></pre>
<pre><code> [1]  1  2  3  4  5  6  7  8  9 10 12 14 16 18 20</code></pre>
<pre class="r"><code># 进行拓扑网络分析
sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = 5)</code></pre>
<pre><code>pickSoftThreshold: will use block size 3600.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 3600 of 3600</code></pre>
<pre><code>Warning: executing %dopar% sequentially: no parallel backend registered</code></pre>
<pre><code>   Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
1      1   0.0278  0.345          0.456  747.00  762.0000 1210.0
2      2   0.1260 -0.597          0.843  254.00  251.0000  574.0
3      3   0.3400 -1.030          0.972  111.00  102.0000  324.0
4      4   0.5060 -1.420          0.973   56.50   47.2000  202.0
5      5   0.6810 -1.720          0.940   32.20   25.1000  134.0
6      6   0.9020 -1.500          0.962   19.90   14.5000   94.8
7      7   0.9210 -1.670          0.917   13.20    8.6800   84.1
8      8   0.9040 -1.720          0.876    9.25    5.3900   76.3
9      9   0.8590 -1.700          0.836    6.80    3.5600   70.5
10    10   0.8330 -1.660          0.831    5.19    2.3800   65.8
11    12   0.8530 -1.480          0.911    3.33    1.1500   58.1
12    14   0.8760 -1.380          0.949    2.35    0.5740   51.9
13    16   0.9070 -1.300          0.970    1.77    0.3090   46.8
14    18   0.9120 -1.240          0.973    1.39    0.1670   42.5
15    20   0.9310 -1.210          0.977    1.14    0.0951   38.7</code></pre>
<pre class="r"><code>str(sft)</code></pre>
<pre><code>List of 2
 $ powerEstimate: num 6
 $ fitIndices   :&#39;data.frame&#39;:  15 obs. of  7 variables:
  ..$ Power         : num [1:15] 1 2 3 4 5 6 7 8 9 10 ...
  ..$ SFT.R.sq      : num [1:15] 0.0278 0.1264 0.3404 0.5062 0.6807 ...
  ..$ slope         : num [1:15] 0.345 -0.597 -1.03 -1.422 -1.716 ...
  ..$ truncated.R.sq: num [1:15] 0.456 0.843 0.972 0.973 0.94 ...
  ..$ mean.k.       : num [1:15] 747 254.5 111 56.5 32.2 ...
  ..$ median.k.     : num [1:15] 761.7 250.8 101.7 47.2 25.1 ...
  ..$ max.k.        : num [1:15] 1206 574 324 202 134 ...</code></pre>
</div>
<div id="可视化" class="section level5">
<h5>2. 可视化</h5>
<p>2.1 sclae-free fit index作为y轴画图 -sign(sft<span class="math inline">\(fitIndices[,3])*sft\)</span>fitIndices[,2]</p>
<pre class="r"><code>cex1 = 0.9
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
xlab=&quot;Soft Threshold (power)&quot;,ylab=&quot;Scale Free Topology Model Fit,signed R^2&quot;,type=&quot;n&quot;,
main = paste(&quot;Scale independence&quot;))

text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
labels=powers,cex=cex1,col=&quot;red&quot;)
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col=&quot;red&quot;)</code></pre>
<img src="figure/02.%20step-by-step%20network%20construction%20and%20module%20detection.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<p>2.2 mean connectivity (degree)作为y轴画图 mean connectivity 对应sft$fitIndices的第五列。</p>
<pre class="r"><code>cex1 = 0.9
plot(sft$fitIndices[,1], sft$fitIndices[,5],
xlab=&quot;Soft Threshold (power)&quot;,ylab=&quot;Mean Connectivity&quot;, type=&quot;n&quot;,
main = paste(&quot;Mean connectivity&quot;))

text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col=&quot;red&quot;)</code></pre>
<img src="figure/02.%20step-by-step%20network%20construction%20and%20module%20detection.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<p>根据上图，选择power = 6，为无尺度网络fit index达到0.9对应的最小power值。</p>
</div>
</div>
<div id="二co-expression-similarity-and-adjacency" class="section level3">
<h3>二、co-expression similarity and adjacency</h3>
<p>计算基因邻近性adjacency，并返回邻近性矩阵。</p>
<pre class="r"><code>softPower = 6
adjacency = adjacency(datExpr, power = softPower) # 返回每个基因对的邻近性
dim(adjacency) # 行和列均对应基因。。。</code></pre>
<pre><code>[1] 3600 3600</code></pre>
<pre class="r"><code>class(adjacency)</code></pre>
<pre><code>[1] &quot;matrix&quot; &quot;array&quot; </code></pre>
<pre class="r"><code>adjacency[1:4, 1:4]</code></pre>
<pre><code>             MMT00000044  MMT00000046  MMT00000051  MMT00000076
MMT00000044 1.000000e+00 2.997850e-09 6.581419e-07 2.279710e-04
MMT00000046 2.997850e-09 1.000000e+00 3.216588e-02 6.390615e-10
MMT00000051 6.581419e-07 3.216588e-02 1.000000e+00 5.254297e-08
MMT00000076 2.279710e-04 6.390615e-10 5.254297e-08 1.000000e+00</code></pre>
</div>
<div id="三topological-overlap-matrixtom" class="section level3">
<h3>三、Topological Overlap Matrix(TOM)</h3>
<p>将邻近矩阵转化为TOM矩阵。 意义：1. 可最大程度降低噪声和虚假关联的影响；2. 见其他资料。</p>
<pre class="r"><code>TOM = TOMsimilarity(adjacency, verbose = F)
dissTOM = 1-TOM
dim(TOM) # 维度同adjacency矩阵。行和列均对应基因</code></pre>
<pre><code>[1] 3600 3600</code></pre>
<pre class="r"><code>TOM[1:4, 1:4]</code></pre>
<pre><code>             [,1]         [,2]         [,3]         [,4]
[1,] 1.0000000000 0.0007954678 6.276637e-04 6.970912e-04
[2,] 0.0007954678 1.0000000000 4.785885e-02 6.144850e-05
[3,] 0.0006276637 0.0478588481 1.000000e+00 7.078494e-05
[4,] 0.0006970912 0.0000614485 7.078494e-05 1.000000e+00</code></pre>
<pre class="r"><code>dissTOM[1:4,1:4]</code></pre>
<pre><code>          [,1]      [,2]      [,3]      [,4]
[1,] 0.0000000 0.9992045 0.9993723 0.9993029
[2,] 0.9992045 0.0000000 0.9521412 0.9999386
[3,] 0.9993723 0.9521412 0.0000000 0.9999292
[4,] 0.9993029 0.9999386 0.9999292 0.0000000</code></pre>
</div>
<div id="四cluster分析" class="section level3">
<h3>四、cluster分析</h3>
<div id="聚类分析及可视化" class="section level5">
<h5>1. 聚类分析及可视化</h5>
<p>输入数据为dissTOM矩阵； 实现函数为fastcluster::hclust【与普通hclust相比，速度更快】</p>
<pre class="r"><code>geneTree = hclust(as.dist(dissTOM), method = &quot;average&quot;);</code></pre>
<p>可视化</p>
<pre class="r"><code>plot(geneTree, xlab=&quot;&quot;, sub=&quot;&quot;, main = &quot;Gene clustering on TOM-based dissimilarity&quot;,
labels = FALSE, hang = 0.04);</code></pre>
<img src="figure/02.%20step-by-step%20network%20construction%20and%20module%20detection.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<p>图说明：一个短竖线——一个leaf——一个基因；branches —— 相互关联、高度共表达的基因簇。</p>
</div>
<div id="module-identification" class="section level5">
<h5>2. module identification</h5>
<p>模块识别等价于识别上图中的独立branches 【cutting the branches off the dendrogram】。 模块识别有多种方法可供选择，这里使用Dynamic Tree Cut（dynamicTreeCut包）的方法。</p>
<p>输入数据为：聚类结果geneTree, 不相似性TOM矩阵dissTOM</p>
<pre class="r"><code># 最小模块基因数：30是一个相对较大的数值。
minModuleSize = 30
# 模块识别
dynamicMods = cutreeDynamic(dendro = geneTree, distM = dissTOM,
deepSplit = 2, pamRespectsDendro = FALSE,
minClusterSize = minModuleSize, verbose = F)
# dynamicMods为一个向量，数值对应基因的模块信息。
dynamicMods[1:4]</code></pre>
<pre><code>   20 20 12 
 0  1  1 21 </code></pre>
<pre class="r"><code>table(dynamicMods)</code></pre>
<pre><code>dynamicMods
  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19 
 88 614 316 311 257 235 225 212 158 153 121 106 102 100  94  91  78  76  65  58 
 20  21  22 
 58  48  34 </code></pre>
<p>结果解读：所有基因共被划分到22个模块，分别标记为1：22，且标号越大，模块越小。此外，0对应未被分配到任何模块的基因。</p>
<p>基因模块识别结果可视化</p>
<pre class="r"><code># 将模块信息映射为颜色信息
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)</code></pre>
<pre><code>dynamicColors
       black         blue        brown         cyan    darkgreen      darkred 
         212          316          311           94           34           48 
       green  greenyellow         grey       grey60    lightcyan   lightgreen 
         235          106           88           76           78           65 
 lightyellow      magenta midnightblue         pink       purple          red 
          58          153           91          158          121          225 
   royalblue       salmon          tan    turquoise       yellow 
          58          100          102          614          257 </code></pre>
<pre class="r"><code># 聚类树和模块信息可视化
plotDendroAndColors(geneTree, dynamicColors, &quot;Dynamic Tree Cut&quot;,
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05,
main = &quot;Gene dendrogram and module colors&quot;)</code></pre>
<img src="figure/02.%20step-by-step%20network%20construction%20and%20module%20detection.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
</div>
</div>
<div id="五合并具有相似表达谱的modules" class="section level3">
<h3>五、合并具有相似表达谱的modules</h3>
<p>Dynamic Tree Cut函数可能会将表达模式非常相似的模块识别为不同模块，因此建议合并这种相似性模块。 实现原理： ①首先计算模块的eigengenes:本质为模块内部基因的PCA分析 ②根据eigengenes之间的相关性对模块聚类， ③合并模块。</p>
<p>①计算模块的eigengenes</p>
<pre class="r"><code>MEList = moduleEigengenes(datExpr, colors = dynamicColors)
MEs = MEList$eigengenes
MEs[1:4, 1:4]</code></pre>
<pre><code>          MEblack        MEblue     MEbrown      MEcyan
F2_2   0.01390248  0.0410177922 0.007072125  0.12978459
F2_3   0.06667534 -0.0009540238 0.072447744 -0.07777835
F2_14  0.06671191 -0.0841292811 0.062700422 -0.19072152
F2_15 -0.06448025  0.0909333146 0.050275810  0.04077621</code></pre>
<pre class="r"><code># 计算模块间的不相似性
MEDiss = 1-cor(MEs)</code></pre>
<p>②根据eigengenes之间的相关性对模块聚类</p>
<pre class="r"><code>METree = hclust(as.dist(MEDiss), method = &quot;average&quot;)
# 结果可视化
plot(METree, main = &quot;Clustering of module eigengenes&quot;,
xlab = &quot;&quot;, sub = &quot;&quot;)</code></pre>
<img src="figure/02.%20step-by-step%20network%20construction%20and%20module%20detection.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<p>③确定阈值，合并模块 确定阈值为0.25，对应相关性系数为0.75。</p>
<pre class="r"><code>MEDissThres = 0.25
# 可视化
plot(METree, main = &quot;Clustering of module eigengenes&quot;,
xlab = &quot;&quot;, sub = &quot;&quot;)
abline(h=MEDissThres, col = &quot;red&quot;)</code></pre>
<img src="figure/02.%20step-by-step%20network%20construction%20and%20module%20detection.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<pre class="r"><code># 合并模块
merge = mergeCloseModules(datExpr, dynamicColors, cutHeight = MEDissThres, verbose = 3)</code></pre>
<pre><code> mergeCloseModules: Merging modules whose distance is less than 0.25
   multiSetMEs: Calculating module MEs.
     Working on set 1 ...
     moduleEigengenes: Calculating 23 module eigengenes in given set.
   multiSetMEs: Calculating module MEs.
     Working on set 1 ...
     moduleEigengenes: Calculating 19 module eigengenes in given set.
   Calculating new MEs...
   multiSetMEs: Calculating module MEs.
     Working on set 1 ...
     moduleEigengenes: Calculating 19 module eigengenes in given set.</code></pre>
<pre class="r"><code># 提取合并后的模块与颜色对应信息
mergedColors = merge$colors;
# 提取合并后的模块的eigengenes信息
mergedMEs = merge$newMEs;</code></pre>
</div>
<div id="其他简单数据探索" class="section level3">
<h3>其他简单数据探索</h3>
<p>对合并前后的模块信息进行可视化</p>
<pre class="r"><code>plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors),
c(&quot;Dynamic Tree Cut&quot;, &quot;Merged dynamic&quot;),
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05)</code></pre>
<img src="figure/02.%20step-by-step%20network%20construction%20and%20module%20detection.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
</div>
<div id="保存数据" class="section level3">
<h3>保存数据</h3>
<pre class="r"><code># Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c(&quot;grey&quot;, standardColors(50));
moduleLabels = match(moduleColors, colorOrder)-1;
MEs = mergedMEs
# Save module colors and labels for use in subsequent parts
save(MEs, moduleLabels, moduleColors, geneTree, file = &quot;./output/FemaleLiver-02-networkConstruction-stepByStep.RData&quot;)</code></pre>
</div>
<div id="总结" class="section level3">
<h3>总结：</h3>
<p>这一步，我们最终获取的信息包括： MEs:合并后的每个模块的epigengene信息 moduleLabels: 合并后的每个基因对应的模块标签信息 moduleLabels: 合并后的每个基因对应的模块颜色信息【WGCNA中用颜色代表模块】 geneTree：基因层级聚类结果</p>
<p>数据格式如下：</p>
<pre class="r"><code>dim(MEs)</code></pre>
<pre><code>[1] 134  19</code></pre>
<pre class="r"><code>MEs[1:4, 1:4]</code></pre>
<pre><code>      MElightgreen     MEblack MEturquoise     MEbrown
F2_2  -0.022004340  0.01390248  0.02035156 0.007072125
F2_3  -0.004099771  0.06667534  0.03761149 0.072447744
F2_14  0.593793143  0.06671191  0.19275250 0.062700422
F2_15 -0.032026159 -0.06448025 -0.02217722 0.050275810</code></pre>
<pre class="r"><code>length(moduleLabels)</code></pre>
<pre><code>[1] 3600</code></pre>
<pre class="r"><code>moduleLabels[1:4]</code></pre>
<pre><code>[1]  0  1  1 21</code></pre>
<pre class="r"><code>length(moduleColors)</code></pre>
<pre><code>[1] 3600</code></pre>
<pre class="r"><code>moduleColors[1:4]</code></pre>
<pre><code>[1] &quot;grey&quot;      &quot;turquoise&quot; &quot;turquoise&quot; &quot;darkred&quot;  </code></pre>
<pre class="r"><code>str(geneTree)</code></pre>
<pre><code>List of 7
 $ merge      : int [1:3599, 1:2] -472 -667 -761 -1831 -1046 -2499 -87 -882 -59 -588 ...
 $ height     : num [1:3599] 0.357 0.364 0.37 0.371 0.372 ...
 $ order      : int [1:3600] 1873 2683 370 878 379 3460 2074 1517 1948 1150 ...
 $ labels     : NULL
 $ method     : chr &quot;average&quot;
 $ call       : language hclust(d = as.dist(dissTOM), method = &quot;average&quot;)
 $ dist.method: NULL
 - attr(*, &quot;class&quot;)= chr &quot;hclust&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] zh_CN.UTF-8/zh_CN.UTF-8/zh_CN.UTF-8/C/zh_CN.UTF-8/zh_CN.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] WGCNA_1.70-3          fastcluster_1.2.3     dynamicTreeCut_1.63-1

loaded via a namespace (and not attached):
  [1] bitops_1.0-7           matrixStats_0.60.0     fs_1.5.0              
  [4] bit64_4.0.5            doParallel_1.0.16      RColorBrewer_1.1-2    
  [7] httr_1.4.2             rprojroot_2.0.2        GenomeInfoDb_1.28.1   
 [10] backports_1.2.1        tools_4.1.0            bslib_0.2.5.1         
 [13] utf8_1.2.2             R6_2.5.0               rpart_4.1-15          
 [16] Hmisc_4.5-0            DBI_1.1.1              BiocGenerics_0.38.0   
 [19] colorspace_2.0-2       nnet_7.3-16            tidyselect_1.1.1      
 [22] gridExtra_2.3          preprocessCore_1.54.0  bit_4.0.4             
 [25] compiler_4.1.0         git2r_0.28.0           Biobase_2.52.0        
 [28] htmlTable_2.2.1        sass_0.4.0             checkmate_2.0.0       
 [31] scales_1.1.1           stringr_1.4.0          digest_0.6.27         
 [34] foreign_0.8-81         rmarkdown_2.10         XVector_0.32.0        
 [37] base64enc_0.1-3        jpeg_0.1-9             pkgconfig_2.0.3       
 [40] htmltools_0.5.1.1      highr_0.9              fastmap_1.1.0         
 [43] htmlwidgets_1.5.3      rlang_0.4.11           impute_1.66.0         
 [46] rstudioapi_0.13        RSQLite_2.2.7          jquerylib_0.1.4       
 [49] generics_0.1.0         jsonlite_1.7.2         dplyr_1.0.7           
 [52] RCurl_1.98-1.3         magrittr_2.0.1         GO.db_3.13.0          
 [55] GenomeInfoDbData_1.2.6 Formula_1.2-4          Matrix_1.3-4          
 [58] Rcpp_1.0.7             munsell_0.5.0          S4Vectors_0.30.0      
 [61] fansi_0.5.0            lifecycle_1.0.0        stringi_1.7.3         
 [64] whisker_0.4            yaml_2.2.1             zlibbioc_1.38.0       
 [67] grid_4.1.0             blob_1.2.2             parallel_4.1.0        
 [70] promises_1.2.0.1       crayon_1.4.1           lattice_0.20-44       
 [73] Biostrings_2.60.2      splines_4.1.0          KEGGREST_1.32.0       
 [76] knitr_1.33             pillar_1.6.2           codetools_0.2-18      
 [79] stats4_4.1.0           glue_1.4.2             evaluate_0.14         
 [82] latticeExtra_0.6-29    data.table_1.14.0      png_0.1-7             
 [85] vctrs_0.3.8            httpuv_1.6.1           foreach_1.5.1         
 [88] gtable_0.3.0           purrr_0.3.4            assertthat_0.2.1      
 [91] cachem_1.0.5           ggplot2_3.3.5          xfun_0.25             
 [94] later_1.2.0            survival_3.2-11        tibble_3.1.3          
 [97] iterators_1.0.13       AnnotationDbi_1.54.1   memoise_2.0.0         
[100] IRanges_2.26.0         workflowr_1.6.2        cluster_2.1.2         
[103] ellipsis_0.3.2        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
