<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Step 3 Relating modules to external information and identifying important genes</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Step 3 Relating modules to external information and identifying important genes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>wgcna/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210814code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210814)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210814code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210814)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong4ffc582"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 4ffc582 </a>
</p>
</div>
<div id="strongRepositoryversionstrong4ffc582" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 4ffc582. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  05.-Network-visualization-using-WGCNA-functions.Rmd
    Untracked:  analysis/03.-Relating-modules-to-external-information-and-identifying-important-genes.Rmd

Unstaged changes:
    Deleted:    CytoscapeInput-edges-brown-red.txt
    Deleted:    CytoscapeInput-nodes-brown-red.txt
    Modified:   Untitled.R
    Deleted:    VisANTInput-brown-top30.txt
    Deleted:    VisANTInput-brown.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/03. Relating modules to external information and identifying important genes.Rmd</code>) and HTML (<code>docs/03. Relating modules to external information and identifying important genes.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
44f102b
</td>
<td>
lily123920
</td>
<td>
2021-08-15
</td>
<td>
change project name
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="加载r包" class="section level3">
<h3>加载R包</h3>
<pre class="r"><code>library(WGCNA)</code></pre>
</div>
<div id="准备数据" class="section level3">
<h3>准备数据</h3>
<pre class="r"><code>lnames1 = load(file = &quot;./data/FemaleLiver-01-dataInput.RData&quot;);
lnames2 &lt;- load(&quot;./data/FemaleLiver-02-networkConstruction-stepByStep.RData&quot;)
lnames1; lnames2</code></pre>
<pre><code>[1] &quot;datExpr&quot;   &quot;datTraits&quot;</code></pre>
<pre><code>[1] &quot;MEs&quot;          &quot;moduleLabels&quot; &quot;moduleColors&quot; &quot;geneTree&quot;    </code></pre>
</div>
<div id="一定量描述module-trait相关性" class="section level3">
<h3>一、定量描述module-trait相关性</h3>
<p>目的：识别与临床traits具有显著相关的modules; 数学原理：计算eigengene和traits数据的相关性，根据相关性大小筛选性状相关模块。 实现过程：输入数据 ##### 1 准备输入数据</p>
<pre class="r"><code># 确定基因和样本数量
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
# 使用module颜色标签重新计算module的eigengenes值。
MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
# tmp &lt;- MEs0[match(rownames(MEs), rownames(MEs0)), match(colnames(MEs), colnames(MEs0))]
# table(tmp == MEs) # TRUE 明明和原来的eigengenes一样，为什么要重新计算呢？？？</code></pre>
<p>至此，已准备好计算相关性分析需要的三个数据： ①MEs: 行名为样本，列名为颜色标记的module，值为epigengene值； ②datTraits：行名为样本（与MEs行名一致），列名为各种临床性状，值为临床性状的值。 ③nSamples：样本数目，为统计学分析所需参数。</p>
<div id="计算相关性" class="section level5">
<h5>2. 计算相关性</h5>
<p>该包只内置了pearson相关性系数的计算。 计算pearson相关性系数及统计显著性（Student asymptotic p值。）</p>
<pre class="r"><code>moduleTraitCor = cor(MEs, datTraits, use = &quot;p&quot;);
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)</code></pre>
</div>
<div id="可视化" class="section level5">
<h5>1.2 可视化</h5>
<p>展示两个信息：pearson相关性系数和统计p值。 颜色——pearson相关性系数； 统计p值通过单元格textlabel显示。</p>
<pre class="r"><code># 准备单元格text信息
textMatrix = paste(signif(moduleTraitCor, 2), &quot;\n(&quot;,
signif(moduleTraitPvalue, 1), &quot;)&quot;, sep = &quot;&quot;)
dim(textMatrix) = dim(moduleTraitCor)

# 热图可视化
labeledHeatmap(Matrix = moduleTraitCor,  # 输入为相关性矩阵
xLabels = colnames(datTraits), # x轴标签
yLabels = colnames(MEs), # y轴标签
ySymbols = colnames(MEs), 
colorLabels = FALSE,
colors = greenWhiteRed(50),
textMatrix = textMatrix,  # 单元格的文字标签
setStdMargins = FALSE,
cex.text = 0.5,
zlim = c(-1,1),
main = paste(&quot;Module-trait relationships&quot;))</code></pre>
<pre><code>Warning in greenWhiteRed(50): WGCNA::greenWhiteRed: this palette is not suitable for people
with green-red color blindness (the most common kind of color blindness).
Consider using the function blueWhiteRed instead.</code></pre>
<img src="figure/03.%20Relating%20modules%20to%20external%20information%20and%20identifying%20important%20genes.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<p>这张图被描述为color-coded table的图形化显示，interesting!其中，行名对应module eigengene；列名对应trait；颜色映射相关性系数；每个单元格包含相关性系数和p值。 【为什么做出来的颜色映射和教程不一致呢？？？】</p>
<p>后续分析，假定weight为我们关注的性状。由上图可知，与weight存在显著正相关的基因模块未MEmagenta和MEpurple。</p>
</div>
</div>
<div id="二gs和mm计算" class="section level3">
<h3>二、GS和MM计算</h3>
<p>GS, gene significance, 评估基因和trait关系； MM, module membership, 评估基因和重要module的关系。</p>
<p>该部分分析为跨模块的分析，与后续”模块内分析“相对。</p>
<div id="准备数据-1" class="section level5">
<h5>1. 准备数据</h5>
<pre class="r"><code># 提取样本的weight数据，并存储为data.frame格式。
weight = as.data.frame(datTraits$weight_g);
colnames(weight) = &quot;weight&quot;

# 清洗module名字格式：如lightgreen
modNames = substring(colnames(MEs), 3)</code></pre>
</div>
<div id="计算mm" class="section level5">
<h5>2. 计算MM</h5>
<p>准备两个data.frame: ①datExpr:3600个基因的表达矩阵 ②MEs:所有module的epigengenes矩阵 两者行名均为样本，顺序一致。</p>
<pre class="r"><code># 计算MM值
geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = &quot;p&quot;))
# 计算p值
MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples))
# 修改列名
colnames(geneModuleMembership) = paste(&quot;MM&quot;, modNames, sep=&quot;&quot;)
names(MMPvalue) = paste(&quot;p.MM&quot;, modNames, sep=&quot;&quot;);</code></pre>
</div>
<div id="计算gs值" class="section level5">
<h5>3. 计算GS值</h5>
<p>准备两个data.frame: ①datExpr:3600个基因的表达矩阵 ②Weight: weight性状的数据 两者行名均为样本，顺序一致。</p>
<pre class="r"><code># 计算相关性
geneTraitSignificance = as.data.frame(cor(datExpr, weight, use = &quot;p&quot;))
# 计算统计p值
GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples))
# 修改行名和列名
names(geneTraitSignificance) = paste(&quot;GS.&quot;, names(weight), sep=&quot;&quot;);
names(GSPvalue) = paste(&quot;p.GS.&quot;, names(weight), sep=&quot;&quot;);</code></pre>
</div>
</div>
<div id="三模块内分析识别高gs和mm基因" class="section level3">
<h3>三、模块内分析：识别高GS和MM基因</h3>
<p>上面已经计算了GS值和MM值。。。这一步只需要提取到相应的数据进行可视化即可。 x轴：MM值：geneModuleMembership[moduleGenes, column]) y轴：GS值：geneTraitSignificance[moduleGenes, 1]) 注意，可视化使用的是绝对值。</p>
<pre class="r"><code>module = &quot;magenta&quot;
column = match(module, modNames)
moduleGenes = moduleColors==module

verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),

                   xlab = paste(&quot;Module Membership in&quot;, module, &quot;module&quot;),

                   ylab = &quot;Gene significance for body weight&quot;,

                   main = paste(&quot;Module membership vs. gene significance\n&quot;),

                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)</code></pre>
<img src="figure/03.%20Relating%20modules%20to%20external%20information%20and%20identifying%20important%20genes.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<p>由上图可知，GS和MM具有强相关性。</p>
</div>
<div id="四网络分析结果汇总" class="section level3">
<h3>四、网络分析结果汇总</h3>
<p>至此，我们已经获得了和weight具有强相关的module，并且识别了该module内的关键基因。 该环节为对上述结果的汇总整理，以方便后续的数据探索。 ###### 1. 查看基因ID类型 该示例教程使用的为probe IDs。</p>
<pre class="r"><code>names(datExpr)[1:6]</code></pre>
<pre><code>[1] &quot;MMT00000044&quot; &quot;MMT00000046&quot; &quot;MMT00000051&quot; &quot;MMT00000076&quot; &quot;MMT00000080&quot;
[6] &quot;MMT00000102&quot;</code></pre>
<pre class="r"><code>names(datExpr)[moduleColors==&quot;magenta&quot;][1:10]</code></pre>
<pre><code> [1] &quot;MMT00000887&quot; &quot;MMT00001077&quot; &quot;MMT00001185&quot; &quot;MMT00001486&quot; &quot;MMT00002002&quot;
 [6] &quot;MMT00002037&quot; &quot;MMT00002102&quot; &quot;MMT00002209&quot; &quot;MMT00002575&quot; &quot;MMT00002758&quot;</code></pre>
<div id="从基因注释信息表提取基因id对应信息" class="section level5">
<h5>2. 从基因注释信息表提取基因ID对应信息</h5>
<pre class="r"><code>annot = read.csv(file = &quot;./data/GeneAnnotation.csv&quot;);
dim(annot)</code></pre>
<pre><code>[1] 23388    34</code></pre>
<pre class="r"><code>names(annot)</code></pre>
<pre><code> [1] &quot;X&quot;                        &quot;ID&quot;                      
 [3] &quot;arrayname&quot;                &quot;substanceBXH&quot;            
 [5] &quot;gene_symbol&quot;              &quot;LocusLinkID&quot;             
 [7] &quot;OfficialGeneSymbol&quot;       &quot;OfficialGeneName&quot;        
 [9] &quot;LocusLinkSymbol&quot;          &quot;LocusLinkName&quot;           
[11] &quot;ProteomeShortDescription&quot; &quot;UnigeneCluster&quot;          
[13] &quot;LocusLinkCode&quot;            &quot;ProteomeID&quot;              
[15] &quot;ProteomeCode&quot;             &quot;SwissprotID&quot;             
[17] &quot;OMIMCode&quot;                 &quot;DirectedTilingPriority&quot;  
[19] &quot;AlternateSymbols&quot;         &quot;AlternateNames&quot;          
[21] &quot;SpeciesID&quot;                &quot;cytogeneticLoc&quot;          
[23] &quot;Organism&quot;                 &quot;clustername&quot;             
[25] &quot;reporterid&quot;               &quot;probeid&quot;                 
[27] &quot;sequenceid&quot;               &quot;clusterid&quot;               
[29] &quot;chromosome&quot;               &quot;startcoordinate&quot;         
[31] &quot;endcoordinate&quot;            &quot;strand&quot;                  
[33] &quot;sequence_3_to_5_prime&quot;    &quot;sequence_5_to_3_prime&quot;   </code></pre>
<pre class="r"><code>probes = colnames(datExpr)
probes2annot = match(probes, annot$substanceBXH) 
sum(is.na(probes2annot)) # Should return 0.</code></pre>
<pre><code>[1] 0</code></pre>
</div>
<div id="构建数据框" class="section level5">
<h5>3. 构建数据框</h5>
<p>该数据框包含所有probeID的以下信息：probe ID, gene symbol, Locus Link ID(Entrez code), module color,；GS for weight, 与所有模块的MM及相应的p值。 并对数据框按照与weight的显著性进行排序，最显著的位于左侧。 小tips:所有信息分别从相应数据框提取；提取顺序与datExpr中的基因顺序一致。</p>
<pre class="r"><code># 构建数据框
geneInfo0 = data.frame(substanceBXH = probes,
geneSymbol = annot$gene_symbol[probes2annot],
LocusLinkID = annot$LocusLinkID[probes2annot],
moduleColor = moduleColors,
geneTraitSignificance,
GSPvalue)
# 按与weight相关性对module进行排序
modOrder = order(-abs(cor(MEs, weight, use = &quot;p&quot;)));
# Add module membership information in the chosen order
for (mod in 1:ncol(geneModuleMembership)){
  oldNames = names(geneInfo0)
  geneInfo0 = data.frame(geneInfo0, 
                         # 
                         geneModuleMembership[, modOrder[mod]], 
                         MMPvalue[, modOrder[mod]])
  
names(geneInfo0) = c(oldNames, paste(&quot;MM.&quot;, modNames[modOrder[mod]], sep=&quot;&quot;),
paste(&quot;p.MM.&quot;, modNames[modOrder[mod]], sep=&quot;&quot;))
}
# 对基因进行排序：首先按照module color,然后按照geneTraitSignificance.
geneOrder = order(geneInfo0$moduleColor, -abs(geneInfo0$GS.weight));
geneInfo = geneInfo0[geneOrder, ]</code></pre>
</div>
</div>
<div id="五保存数据" class="section level3">
<h3>五、保存数据</h3>
<pre class="r"><code>write.csv(geneInfo, file = &quot;./output/geneInfo.csv&quot;)</code></pre>
</div>
<div id="总结" class="section level3">
<h3>总结</h3>
<p>至此，我们得到一个矩阵，包含了3600个基因的相关汇总信息。包括： 1. 基因所处的模块信息； 2. 基因在模块内的权重； 3. 基因与性状的关联程度。</p>
<p>包含信息如下：</p>
<pre class="r"><code>colnames(geneInfo)</code></pre>
<pre><code> [1] &quot;substanceBXH&quot;      &quot;geneSymbol&quot;        &quot;LocusLinkID&quot;      
 [4] &quot;moduleColor&quot;       &quot;GS.weight&quot;         &quot;p.GS.weight&quot;      
 [7] &quot;MM.magenta&quot;        &quot;p.MM.magenta&quot;      &quot;MM.purple&quot;        
[10] &quot;p.MM.purple&quot;       &quot;MM.midnightblue&quot;   &quot;p.MM.midnightblue&quot;
[13] &quot;MM.green&quot;          &quot;p.MM.green&quot;        &quot;MM.black&quot;         
[16] &quot;p.MM.black&quot;        &quot;MM.turquoise&quot;      &quot;p.MM.turquoise&quot;   
[19] &quot;MM.cyan&quot;           &quot;p.MM.cyan&quot;         &quot;MM.blue&quot;          
[22] &quot;p.MM.blue&quot;         &quot;MM.grey60&quot;         &quot;p.MM.grey60&quot;      
[25] &quot;MM.lightcyan&quot;      &quot;p.MM.lightcyan&quot;    &quot;MM.royalblue&quot;     
[28] &quot;p.MM.royalblue&quot;    &quot;MM.grey&quot;           &quot;p.MM.grey&quot;        
[31] &quot;MM.darkgreen&quot;      &quot;p.MM.darkgreen&quot;    &quot;MM.pink&quot;          
[34] &quot;p.MM.pink&quot;         &quot;MM.salmon&quot;         &quot;p.MM.salmon&quot;      
[37] &quot;MM.greenyellow&quot;    &quot;p.MM.greenyellow&quot;  &quot;MM.lightgreen&quot;    
[40] &quot;p.MM.lightgreen&quot;   &quot;MM.darkred&quot;        &quot;p.MM.darkred&quot;     
[43] &quot;MM.brown&quot;          &quot;p.MM.brown&quot;       </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] zh_CN.UTF-8/zh_CN.UTF-8/zh_CN.UTF-8/C/zh_CN.UTF-8/zh_CN.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] WGCNA_1.70-3          fastcluster_1.2.3     dynamicTreeCut_1.63-1

loaded via a namespace (and not attached):
  [1] bitops_1.0-7           matrixStats_0.60.0     fs_1.5.0              
  [4] bit64_4.0.5            doParallel_1.0.16      RColorBrewer_1.1-2    
  [7] httr_1.4.2             rprojroot_2.0.2        GenomeInfoDb_1.28.1   
 [10] backports_1.2.1        tools_4.1.0            bslib_0.2.5.1         
 [13] utf8_1.2.2             R6_2.5.0               rpart_4.1-15          
 [16] Hmisc_4.5-0            DBI_1.1.1              BiocGenerics_0.38.0   
 [19] colorspace_2.0-2       nnet_7.3-16            tidyselect_1.1.1      
 [22] gridExtra_2.3          preprocessCore_1.54.0  bit_4.0.4             
 [25] compiler_4.1.0         git2r_0.28.0           Biobase_2.52.0        
 [28] htmlTable_2.2.1        sass_0.4.0             checkmate_2.0.0       
 [31] scales_1.1.1           stringr_1.4.0          digest_0.6.27         
 [34] foreign_0.8-81         rmarkdown_2.10         XVector_0.32.0        
 [37] base64enc_0.1-3        jpeg_0.1-9             pkgconfig_2.0.3       
 [40] htmltools_0.5.1.1      highr_0.9              fastmap_1.1.0         
 [43] htmlwidgets_1.5.3      rlang_0.4.11           impute_1.66.0         
 [46] rstudioapi_0.13        RSQLite_2.2.7          jquerylib_0.1.4       
 [49] generics_0.1.0         jsonlite_1.7.2         dplyr_1.0.7           
 [52] RCurl_1.98-1.3         magrittr_2.0.1         GO.db_3.13.0          
 [55] GenomeInfoDbData_1.2.6 Formula_1.2-4          Matrix_1.3-4          
 [58] Rcpp_1.0.7             munsell_0.5.0          S4Vectors_0.30.0      
 [61] fansi_0.5.0            lifecycle_1.0.0        stringi_1.7.3         
 [64] whisker_0.4            yaml_2.2.1             zlibbioc_1.38.0       
 [67] grid_4.1.0             blob_1.2.2             parallel_4.1.0        
 [70] promises_1.2.0.1       crayon_1.4.1           lattice_0.20-44       
 [73] Biostrings_2.60.2      splines_4.1.0          KEGGREST_1.32.0       
 [76] knitr_1.33             pillar_1.6.2           codetools_0.2-18      
 [79] stats4_4.1.0           glue_1.4.2             evaluate_0.14         
 [82] latticeExtra_0.6-29    data.table_1.14.0      png_0.1-7             
 [85] vctrs_0.3.8            httpuv_1.6.1           foreach_1.5.1         
 [88] gtable_0.3.0           purrr_0.3.4            assertthat_0.2.1      
 [91] cachem_1.0.5           ggplot2_3.3.5          xfun_0.25             
 [94] later_1.2.0            survival_3.2-11        tibble_3.1.3          
 [97] iterators_1.0.13       AnnotationDbi_1.54.1   memoise_2.0.0         
[100] IRanges_2.26.0         workflowr_1.6.2        cluster_2.1.2         
[103] ellipsis_0.3.2        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
